src/test/regress/expected/multi_partitioning.out

```diff
@@ -1584,27 +1584,25 @@ SELECT success FROM run_command_on_workers('select pg_reload_conf()');
 
 EXPLAIN (COSTS OFF)
 SELECT * FROM partitioning_hash_test JOIN partitioning_hash_join_test USING (id, subid);
-                                                                                QUERY PLAN
+                                                                           QUERY PLAN
 ---------------------------------------------------------------------
  Custom Scan (Citus Adaptive)
    Task Count: 4
    Tasks Shown: One of 4
    ->  Task
          Node: host=localhost port=xxxxx dbname=regression
-         ->  Gather
```

Comment:
Is this test flappy now? Otherwise we should enable it before.

If it is flappy now, we can remove the query below that disables parallel, because this PR does it by default now.

Reply:
I don't follow. We're disabling max_parallel_workers_per_gather for all tests to prevent it from spuriously showing up when not expected or not showing up when expected.

In this test, it usually shows up, but sometimes doesn't, because you can't easily force parallelism. Now it never does and is therefore no longer flappy.

Reply:
Hmz, seems we've added a bunch of tests that rely on max_parallel_workers_per_gather, which is odd because you cannot rely on it at all.

Removing them.

Reply:
Actually, the one in multi_distributed_transaction_id.sql  has a super good reason for existing, keeping that one.

Reply:
Sorry yes, I meant for any tests that actually test that parallel behaviour works we should change `max_parallel_workers_per_gather` back again.